### ISSUE TITLE
Add shell mode, allwo -p to point to an application in PATH instead of absolute path.

### ISSUE THREAD

**Issue by TheOneRing** [chocolatey/shimgen#38]
Examples:
shimgen.exe -p powershell -c "Write-Host 'Hello World'" ...

**Comment by ferventcoder** [comment]
Would you see this as it finding that absolute path or just running shellex?

**Comment by TheOneRing** [comment]
as shellex,

it would allow providing a nice looking and relocatable wrapper for scripts.
For example a special invocation of python, currently such a shim would not be portable as we need to hardcode the path to python

**Comment by ferventcoder** [comment]
Yeah, there is code in Chocolatey now to handle these types of things - look in https://github.com/chocolatey/choco and search for DotNetFileSystem. We have a special method in there for resolving binary paths from just the name.
